lsp-mode.el

```diff
@@ -5671,20 +5672,25 @@ while PLACEHOLDER?, is either nil or a string suggested by the
 language server as the initial input of a new-name prompt."
   (unless (lsp-feature? "textDocument/rename")
     (error "The connected server(s) doesn't support renaming"))
-  (if (and lsp-rename-use-prepare (lsp-feature? "textDocument/prepareRename"))
-      (when-let ((response
-                  (lsp-request "textDocument/prepareRename"
-                               (lsp--text-document-position-params))))
-        (let* ((bounds (lsp--range-to-region
-                        (if (lsp-range? response)
-                            response
-                          (lsp:prepare-rename-result-range response))))
-               (placeholder
-                (and (not (lsp-range? response))
-                     (lsp:prepare-rename-result-placeholder response))))
-          (cons bounds placeholder)))
-    (when-let ((bounds (bounds-of-thing-at-point 'symbol)))
-      (cons bounds nil))))
+  (or
+   (and lsp-rename-use-prepare (lsp-feature? "textDocument/prepareRename")
+        (if-let ((response (lsp-request "textDocument/prepareRename"
+                                        (lsp--text-document-position-params))))
+            (unless (lsp:prepare-rename-default-behavior-result-default-behavior
```

Comment:
Why don't you just check type `lsp-prepare-rename-default-behavior-result?` instead?